tests/eth_general.py

```diff
@@ -919,6 +916,28 @@ def will_decode_instruction_callback(self, state, pc):
         context = p.context.get('flags', {})
         self.assertTrue(context.get('found', False))
 
+    def test_preconstraints(self):
+        source_code = '''
+        contract C {
+            constructor() public {}
+            function f0() public {}
+            function f1(uint a) public payable {}
+        }
+        '''
+        m: ManticoreEVM = self.mevm
+
+        creator_account = m.create_account(balance=1000)
+        contract_account = m.solidity_create_contract(source_code, owner=creator_account, balance=0)
+
+        data = m.make_symbolic_buffer(320)
+        value = m.make_symbolic_value()
+        m.constrain(m.preconstraint_for_call_transaction(address=contract_account, data=data, value=value))
+        m.transaction(caller=creator_account, address=contract_account, data=data, value=value)
+
+        results = [state.platform.all_transactions[-1].result for state in m.all_states]
+        # The TXERROR indicates a state where the sent value is greater than the senders budget.
+        self.assertListEqual(sorted(results), ['STOP']*2 + ['TXERROR'])
```

Comment:
Is there a possibility to assert applied constraints?

Reply:
I could try to pattern match the constraint returned from `preconstraint_for_call_transaction`, but that would be a lot of effort and would make the test sensitive to implementation details of the constraint system. If the constraint was constructed wrongly, the state results would very likely include a `REVERT` or miss one of the items that the `assertListEqual` looks for.

Reply:
Ok, that makes sense. :+1: